### ISSUE TITLE
systemd-networkd: doesn't remove "links" file

### ISSUE THREAD

**Issue by kayrus** [coreos/bugs#1081]
Every time you run a container, systemd-networkd creates a file inside `/run/systemd/netif/links`. Unfortunately systemd-networkd doesn't remove these files when container exits. That could case `/run` tmpfs oversize.

It was fixed in systemd v228. https://github.com/systemd/systemd/commit/84de38c56915e14c148f558c6acc489a00755696#diff-12a3ecdf5af7aa9f8583c27ac442ca43R380

Possibly we can backport this patch and add only one string to avoid this behavior:

```diff
diff --git a/src/network/networkd-link.c b/src/network/networkd-link.c
index cc9dc39..b3e2101 100644
--- a/src/network/networkd-link.c
+++ b/src/network/networkd-link.c
@@ -315,6 +315,8 @@ static void link_free(Link *link) {
         if (link->manager)
                 hashmap_remove(link->manager->links, INT_TO_PTR(link->ifindex));
 
+        (void) unlink(link->state_file);
+
         free(link->ifname);
 
         free(link->state_file);
```

This fix works fine in my environment.

/cc @crawford @marineam @mischief

**Comment by lrvick** [comment]
This fills the /run tmpfs on many of my hosts, and also brought a production etcd cluster to a complete halt when systemd-networkd could no longer function on 3/5 hosts.

Can easily reproduce on 835.9:

```
core@ip-10-10-12-185 ~ $ docker run --rm -it debian:jessie echo && ls /run/systemd/netif/links/ | wc -l
42587
core@ip-10-10-12-185 ~ $ docker run --rm -it debian:jessie echo && ls /run/systemd/netif/links/ | wc -l
42589
core@ip-10-10-12-185 ~ $ docker run --rm -it debian:jessie echo && ls /run/systemd/netif/links/ | wc -l
42591
```

**Comment by mischief** [comment]
if anyone else runs into this before the patch is applied, a workaround is:

```
sudo rm -r /run/systemd/netif/links && sudo systemctl restart systemd-networkd
```

**Comment by mischief** [comment]
fixed by https://github.com/coreos/coreos-overlay/pull/1790

**Comment by crawford** [comment]
This change introduced regressions, so it was rolled back.

**Comment by kayrus** [comment]
Is it reproducible in vanilla systemd v228?

**Comment by mischief** [comment]
fixed by coreos/coreos-overlay#1833
